src/terminal.ts

```diff
@@ -1,30 +1,49 @@
-import * as vscode from "vscode";
+import * as vs from "vscode";
 
-export const ensureTerminalExists = (): boolean => {
-  if (vscode.window.terminals.length === 0) {
-    createNewTerminal();
+export const ensureTerminalExists = (cmdName: string = 'Makefile Runner'): boolean => {
```

Comment:
Now an optional name for each new terminal, but all code paths currently pass the command name as the name of the new terminal.